cmake.deps/cmake/BuildLibuv.cmake

```
@@ -1,77 +1,19 @@
-    "TARGET"
-    "PATCH_COMMAND;CONFIGURE_COMMAND;BUILD_COMMAND;INSTALL_COMMAND"
-    ${ARGN})
-
-  if(NOT _libuv_CONFIGURE_COMMAND AND NOT _libuv_BUILD_COMMAND
-        AND NOT _libuv_INSTALL_COMMAND)
-    message(FATAL_ERROR "Must pass at least one of CONFIGURE_COMMAND, BUILD_COMMAND, INSTALL_COMMAND")
-  endif()
-  if(NOT _libuv_TARGET)
-    set(_libuv_TARGET "libuv")
-  endif()
-
-  ExternalProject_Add(${_libuv_TARGET}
-    PREFIX ${DEPS_BUILD_DIR}
-    URL ${LIBUV_URL}
-    DOWNLOAD_DIR ${DEPS_DOWNLOAD_DIR}/libuv
-    DOWNLOAD_COMMAND ${CMAKE_COMMAND}
-      -DPREFIX=${DEPS_BUILD_DIR}
-      -DDOWNLOAD_DIR=${DEPS_DOWNLOAD_DIR}/libuv
-      -DURL=${LIBUV_URL}
-      -DEXPECTED_SHA256=${LIBUV_SHA256}
-      -DTARGET=${_libuv_TARGET}
-      -DUSE_EXISTING_SRC_DIR=${USE_EXISTING_SRC_DIR}
-      -P ${CMAKE_CURRENT_SOURCE_DIR}/cmake/DownloadAndExtractFile.cmake
-    PATCH_COMMAND "${_libuv_PATCH_COMMAND}"
-    BUILD_IN_SOURCE ${_libuv_BUILD_IN_SOURCE}
-    CONFIGURE_COMMAND "${_libuv_CONFIGURE_COMMAND}"
-    BUILD_COMMAND "${_libuv_BUILD_COMMAND}"
-    INSTALL_COMMAND "${_libuv_INSTALL_COMMAND}")
-endfunction()
-
-set(UNIX_CFGCMD sh ${DEPS_BUILD_DIR}/src/libuv/autogen.sh &&
-  ${DEPS_BUILD_DIR}/src/libuv/configure --with-pic --disable-shared
-  --prefix=${DEPS_INSTALL_DIR} --libdir=${DEPS_INSTALL_DIR}/lib
-  CC=${DEPS_C_COMPILER})
-
-if(UNIX)
-  BuildLibuv(
-    CONFIGURE_COMMAND ${UNIX_CFGCMD} MAKE=${MAKE_PRG}
-    INSTALL_COMMAND ${MAKE_PRG} V=1 install)
-
-elseif(WIN32)
-
-  set(UV_OUTPUT_DIR ${DEPS_BUILD_DIR}/src/libuv/${CMAKE_BUILD_TYPE})
-  if(MSVC)
-    set(BUILD_SHARED ON)
-  elseif(MINGW)
-    set(BUILD_SHARED OFF)
-  else()
-    message(FATAL_ERROR "Trying to build libuv in an unsupported system ${CMAKE_SYSTEM_NAME}/${CMAKE_C_COMPILER_ID}")
-  endif()
-  BuildLibUv(BUILD_IN_SOURCE
-    CONFIGURE_COMMAND ${CMAKE_COMMAND} -E copy
-        ${CMAKE_CURRENT_SOURCE_DIR}/cmake/LibuvCMakeLists.txt
-        ${DEPS_BUILD_DIR}/src/libuv/CMakeLists.txt
-      COMMAND ${CMAKE_COMMAND} ${DEPS_BUILD_DIR}/src/libuv/CMakeLists.txt
-        -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
-        -DCMAKE_GENERATOR=${CMAKE_GENERATOR}
-        -DCMAKE_GENERATOR_PLATFORM=${CMAKE_GENERATOR_PLATFORM}
-        -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
-        -DBUILD_SHARED_LIBS=${BUILD_SHARED}
-        -DCMAKE_INSTALL_PREFIX=${DEPS_INSTALL_DIR}
-    PATCH_COMMAND ${LIBUV_PATCH_COMMAND}
-    BUILD_COMMAND ${CMAKE_COMMAND} --build . --config ${CMAKE_BUILD_TYPE}
-    INSTALL_COMMAND ${CMAKE_COMMAND} --build . --target install --config ${CMAKE_BUILD_TYPE})
-
-else()
-  message(FATAL_ERROR "Trying to build libuv in an unsupported system ${CMAKE_SYSTEM_NAME}/${CMAKE_C_COMPILER_ID}")
-endif()
+ExternalProject_Add(libuv
+  PREFIX ${DEPS_BUILD_DIR}
+  URL ${LIBUV_URL}
+  CMAKE_ARGS
+    -DCMAKE_INSTALL_PREFIX=${DEPS_INSTALL_DIR}
+    -DCMAKE_INSTALL_LIBDIR=lib
+    -DBUILD_TESTING=OFF
+    -DCMAKE_POSITION_INDEPENDENT_CODE=ON
+  DOWNLOAD_DIR ${DEPS_DOWNLOAD_DIR}/libuv
```

Instead of adjusting the glob, you can also use this:
```suggestion
    -DCMAKE_POSITION_INDEPENDENT_CODE=ON
    -DCMAKE_INSTALL_LIBDIR=lib
  DOWNLOAD_DIR ${DEPS_DOWNLOAD_DIR}/libuv
```
Adjusting the glob might lead to it someday picking stuff up that you don't want.

---

Haha, I already did that. I can remove the glob now.

Pretty sure the shared lib being built is screwing up luv.

---

Oh, oops. Didn't see 😅